app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt

```diff
@@ -1,46 +1,316 @@
+  val account =
+      UserAccount(
+          "",
+          "Micheal",
+          "Phelps",
+          1.8f,
+          HeightUnit.METER,
+          70f,
+          WeightUnit.KG,
+          Gender.MALE,
+          Date(),
+          "")
+  val profile = R.drawable.homme
+
+  // Temp until got real workouts repo
+  val workouts =
+      listOf(
+          BodyWeightWorkout(
+              workoutId = "1",
+              name = "Run in Lavaux",
+              description = "Enjoying Lavaux with user2 and user3",
+              warmup = true,
+              userIdSet = mutableSetOf("user1", "user2", "user3")),
+          YogaWorkout(
+              workoutId = "3",
+              name = "After Comparch relax",
+              description = "Chilling time",
+              warmup = true,
+              userIdSet = mutableSetOf("user1")),
+          BodyWeightWorkout(
+              workoutId = "2",
+              name = "Summer body",
+              description = "Be ready for the summer",
+              warmup = false,
+              userIdSet = mutableSetOf("user1", "user4")),
+      )
+
   Scaffold(
-      modifier = Modifier.testTag("mainScreen"),
-      topBar = {
-        TopAppBar(
-            title = { Text("Main Screen") },
-            actions = {
-              IconButton(onClick = { navigationActions.navigateTo("Settings Screen") }) {
-                Icon(
-                    imageVector = Icons.Outlined.Settings,
-                    contentDescription = "Settings",
-                    tint = Color.Black)
-              }
-            })
-      },
+      modifier = Modifier.testTag("MainScreen"),
       content = { padding ->
-        Box(
+        Column(
             modifier = Modifier.fillMaxSize().padding(padding),
-            contentAlignment = Alignment.Center) {
-              Text("No content yet")
+            verticalArrangement = Arrangement.SpaceBetween) {
+              // Display the profile section
+              ProfileSection(
+                  account = account, profile = profile, navigationActions = navigationActions)
+
+              // Display the workout sessions
+              WorkoutSessionsSection(
+                  workouts = workouts, profile = profile, navigationActions = navigationActions)
+
+              // Display the quick workout section
+              QuickWorkoutSection(navigationActions = navigationActions)
+
+              // Display the new workout plan button
+              NewWorkoutSection(navigationActions = navigationActions)
             }
       },
-      bottomBar = {
-        BottomNavigationMenu(
-            onTabSelect = { route -> navigationActions.navigateTo(route) },
-            tabList = LIST_OF_TOP_LEVEL_DESTINATIONS,
-            selectedItem = navigationActions.currentRoute())
-      })
+      bottomBar = { BottomNavigationBar(navigationActions = navigationActions) })
+}
+
+// Composable function for the profile section with profile picture and settings
```

Comment:
It is very good that you thought of adding comments. I would suggest using the KDoc format (it is similar to JavaDoc).

Reply:
Good idea I will do it

Reply:
Good idea I will do it.